components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -432,63 +442,300 @@ public User updateUser(User user, Map<String, Boolean> requiredAttributes) throw
+     *
+     * @param node
+     * @param requiredAttributes
+     * @param offset
+     * @param limit
+     * @param sortBy
+     * @param sortOrder
+     * @param domainName
+     * @param filteredUsers
+     * @return
+     * @throws CharonException
+     */
+    private List<Object> getMultiAttributeFilteredUsers(Node node, Map<String, Boolean> requiredAttributes,
+                                                        int offset, int limit, String sortBy, String sortOrder,
+                                                        String domainName, List<Object> filteredUsers)
+            throws CharonException {
+
+        String[] userNames;
+
+        // Handle pagination.
+        if (limit > 0) {
+            userNames = getFilteredUsersFromMultiAttributeFiltering(node, offset, limit, sortBy,
+                    sortOrder, domainName);
+            filteredUsers.set(0, userNames.length);
+            filteredUsers.addAll(getFilteredUserDetails(userNames, requiredAttributes));
+        } else {
+            int maxLimit = getMaxLimit();
+            if (StringUtils.isNotEmpty(domainName)) {
+                userNames = getFilteredUsersFromMultiAttributeFiltering(node, offset, maxLimit, sortBy,
+                        sortOrder, domainName);
+                filteredUsers.set(0, userNames.length);
+                filteredUsers.addAll(getFilteredUserDetails(userNames, requiredAttributes));
+            } else {
+                int totalUserCount = 0;
+                // If pagination and domain name are not given, then perform filtering on all available user stores.
+                while (carbonUM != null) {
+                    // If carbonUM is not an instance of Abstract User Store Manger we can't get the domain name.
+                    if (carbonUM instanceof AbstractUserStoreManager) {
+                        domainName = carbonUM.getRealmConfiguration().getUserStoreProperty("DomainName");
+                        userNames = getFilteredUsersFromMultiAttributeFiltering(node, offset, maxLimit,
+                                sortBy, sortOrder, domainName);
+                        totalUserCount += userNames.length;
+                        filteredUsers.addAll(getFilteredUserDetails(userNames, requiredAttributes));
+                    }
+                    carbonUM = carbonUM.getSecondaryUserStoreManager();
                 }
-                return filteredUsers;
+                //set the total results
+                filteredUsers.set(0, totalUserCount);
+            }
+        }
+        return filteredUsers;
+    }
+
+    /**
+     * Get maximum user limit to retrieve.
+     *
+     * @return
+     */
+    private int getMaxLimit() {
+
+        int givenMax;
+
+        try {
+            givenMax = Integer.parseInt(carbonUM.getRealmConfiguration().getUserStoreProperty(
+                    "MaxUserNameListLength"));
+        } catch (Exception e) {
```

Comment:
Why catching genric Excpetion? try to catch specific exceptions

Reply:
Outdated codes